$OpenBSD: patch-Network_Socket_hsc,v 1.5 2014/11/22 20:48:00 kili Exp $

don't throw an error when attempting to setsockopt IPV6_V6ONLY

--- Network/Socket.hsc.orig	Fri Apr 18 09:16:03 2014
+++ Network/Socket.hsc	Wed Nov 12 22:40:00 2014
@@ -327,9 +327,10 @@ socket family stype protocol = do
     socket_status <- newMVar NotConnected
     let sock = MkSocket fd family stype protocol socket_status
 #if HAVE_DECL_IPV6_V6ONLY
-# if defined(mingw32_HOST_OS)
+# if defined(mingw32_HOST_OS) || defined(openbsd_HOST_OS)
     -- the IPv6Only option is only supported on Windows Vista and later,
-    -- so trying to change it might throw an error
+    -- so trying to change it might throw an error. It's also not
+    -- supported on OpenBSD.
     when (family == AF_INET6) $
             E.catch (setSocketOption sock IPv6Only 0) $ (\(_ :: E.IOException) -> return ())
 # else
